src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigDiscoveryVisitor.java

```diff
@@ -22,27 +25,100 @@ public JoshConfigDiscoveryVisitor() {
   }
 
   @Override
-  public Set<String> visitConfigValue(JoshLangParser.ConfigValueContext ctx) {
+  public Set<DiscoveredConfigVar> visitConfigValue(JoshLangParser.ConfigValueContext ctx) {
     String variableName = ctx.name.getText();
-    Set<String> result = new HashSet<>();
-    result.add(variableName);
-    
+    DiscoveredConfigVar discovered = new DiscoveredConfigVar(variableName);
+    Set<DiscoveredConfigVar> result = new HashSet<>();
+    result.add(discovered);
+
     // Continue visiting children to collect any nested config values
-    Set<String> childResult = visitChildren(ctx);
+    Set<DiscoveredConfigVar> childResult = visitChildren(ctx);
     if (childResult != null) {
       result.addAll(childResult);
     }
-    
+
     return result;
   }
 
   @Override
-  protected Set<String> defaultResult() {
+  public Set<DiscoveredConfigVar> visitConfigValueWithDefault(JoshLangParser.ConfigValueWithDefaultContext ctx) {
+    String variableName = ctx.name.getText();
+    String defaultValueString = extractDefaultValueString(ctx.defaultValue);
+    DiscoveredConfigVar discovered = new DiscoveredConfigVar(variableName, defaultValueString);
+    Set<DiscoveredConfigVar> result = new HashSet<>();
+    result.add(discovered);
+
+    // Continue visiting children to collect any nested config values
+    Set<DiscoveredConfigVar> childResult = visitChildren(ctx);
+    if (childResult != null) {
+      result.addAll(childResult);
+    }
+
+    return result;
+  }
+
+  /**
+   * Extracts a readable string representation from a default value expression.
+   *
+   * <p>This method traverses the parse tree to build a human-readable string
+   * representation of the default value, handling common cases like numbers
+   * with units, strings, and simple expressions.</p>
+   *
+   * @param defaultValueContext The parse tree context for the default value expression
+   * @return A string representation of the default value
+   */
+  private String extractDefaultValueString(ParseTree defaultValueContext) {
+    if (defaultValueContext == null) {
+      return "";
+    }
+
+    // Handle simple cases based on context type
+    if (defaultValueContext instanceof JoshLangParser.SimpleExpressionContext) {
```

Comment:
Don't love this instanceof. Is there another option in ANTLR? Also, can we switch to new style switch statement?